### PR TITLE
Feature/book reservation form

### LINKED ISSUE #36: Book reservation form
Look at #2

### PR DESCRIPTION
**Pull request by victorsnunes**
Implements book reservation form in #2

### CODE REVIEW

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+          print(
+              formattedDate); //formatted date output using intl package =>  2021-03-16
+          //you can implement different kind of Date Format here according to your requirement
+
+          setState(() {
+            begin_date_controller.text =
+                formattedDate; //set output date to TextField value.
+          });
+        } else {
+          print("Date is not selected");
```
Comment (BrunoRosendo) [inline src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Are these prints supposed to be here?

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+              formattedDate); //formatted date output using intl package =>  2021-03-16
+          //you can implement different kind of Date Format here according to your requirement
+
+          setState(() {
+            end_date_controller.text =
+                formattedDate; //set output date to TextField value.
+          });
+        } else {
+          print("Date is not selected");
+        }
```
Comment (BrunoRosendo) [inline src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Same thing with the prints

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+              if (!_formKey.currentState.validate()) {
+                return;
+              }
+
+              _formKey.currentState.save();
+
+              print(_begin_date);
+              print(_end_date);
+              print(_is_urgent);
+
```
Comment (BrunoRosendo) [inline src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Same thing with the prints

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        ),
+        actions: [
+          TextButton(
+            child: Text('SUBMIT'),
+            onPressed: () {
+              if (!_formKey.currentState.validate()) {
+                return;
+              }
+
+              _formKey.currentState.save();
```
Comment (BrunoRosendo) [inline src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: What does this `save` do? I'm curious, as I didn't use it in other forms

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                ),
+          ),
+        ),
+        actions: [
+          TextButton(
+            child: Text('SUBMIT'),
+            onPressed: () {
+              if (!_formKey.currentState.validate()) {
+                return;
+              }
```
Comment (BrunoRosendo) [inline src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: We should maybe have 2 buttons (submit and cancel), the same way we have in the [mockup](https://github.com/LEIC-ES-2021-22/3LEIC03T1/issues/2) and in the [search filters](https://github.com/LEIC-ES-2021-22/3LEIC03T1/blob/2d0551d5bf4582b6c60e154ad749426f79fd7868/src/app_feup/lib/view/Widgets/search_filter_form.dart#L78)

The elevated button is the one with the red background

Comment (BrunoRosendo) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Also, we should have it in portuguese, since the app is all the in portuguese

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]
This is looking pretty good. I left some comments

Apart from that, please try to squash the commits in the way that the commit history is more readable and effectively tells us what was implemented in this task (telling a story, like the teacher said).
You can do this by using `git rebase -i <commit>` where commit is the hash of the last commit you want to mantain. When done, you need `git push --force-with-lease` to update this PR

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 4 above)
Comment (victorsnunes) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Good question hahah

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 5 above)
Comment (victorsnunes) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Right, I'll do it

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 4 above)
Comment (BrunoRosendo) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: I think it's supposed to save the form? Can you check if the state is reset after you re-open the dialog? I'm afraid the state will persist after we close it

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 5 above)
Comment (BrunoRosendo) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: This is not changed yet

**Comment by BrunoRosendo** [comment]
@victorsnunes Please start marking the comments as resolved when you solve them, it becomes much easier to manage. Also, please look at what I wrote about the commits and the rebase. If you want, we can help you with that

**Comment by ruialves35** [comment]
Any improvement on this task? As @BrunoRosendo said, don't forget to close the comments that you already solved @victorsnunes , it's much easier to check the development of this

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 5 above)
Comment (laureansvictor) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: @BrunoRosendo something like that?

Screenshot from 2022-05-30 23-37-34

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 4 above)
Comment (laureansvictor) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: @BrunoRosendo  @victorsnunes  was this fixed?

Comment (BrunoRosendo) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: I think só. Its working as intended

File: src/app_feup/lib/view/Widgets/book_reservation_dialog.dart (hunk 5 above)
Comment (BrunoRosendo) [reply src/app_feup/lib/view/Widgets/book_reservation_dialog.dart]: Yes! But please be consistent with other pages and:
- Put the cancel on the left
- Change the submit button to the app's main theme (please see search filters for reference)
- Change the button texts to Portuguese

**Comment by BrunoRosendo** [comment]
Just missing a rebase I think?

**Comment by victorsnunes** [comment]
> Just missing a rebase I think?
I merged the master on this branch, and it says it's already up to date.

**Review by BrunoRosendo — APPROVED** [review]

**Pull request merged**